scripts/main/mapview.js

```diff
@@ -174,43 +230,56 @@ mapview.open = function (albumID = null) {
 		}
 	};
 
-	// Adds photos to the map
+	/**
+	 * Adds photos to the map.
+	 *
+	 * @param {(Album|TagAlbum|PositionData)} album
+	 */
 	const addPhotosToMap = function (album) {
 		// check if empty
 		if (!album.photos) return;
 
+		/** @type {MapPhotoEntry[]} */
 		let photos = [];
 
-		album.photos.forEach(function (element, index) {
-			if (element.latitude || element.longitude) {
-				photos.push({
-					lat: parseFloat(element.latitude),
-					lng: parseFloat(element.longitude),
-					thumbnail: element.sizeVariants.thumb !== null ? element.sizeVariants.thumb.url : "img/placeholder.png",
-					thumbnail2x: element.sizeVariants.thumb2x !== null ? element.sizeVariants.thumb2x.url : null,
-					url: element.sizeVariants.small !== null ? element.sizeVariants.small.url : element.url,
-					url2x: element.sizeVariants.small2x !== null ? element.sizeVariants.small2x.url : null,
-					name: element.title,
-					taken_at: element.taken_at,
-					albumID: element.album,
-					photoID: element.id,
-				});
-
-				// Update min/max lat/lng
-				if (mapview.min_lat === null || mapview.min_lat > element.latitude) {
-					mapview.min_lat = parseFloat(element.latitude);
-				}
-				if (mapview.min_lng === null || mapview.min_lng > element.longitude) {
-					mapview.min_lng = parseFloat(element.longitude);
-				}
-				if (mapview.max_lat === null || mapview.max_lat < element.latitude) {
-					mapview.max_lat = parseFloat(element.latitude);
-				}
-				if (mapview.max_lng === null || mapview.max_lng < element.longitude) {
-					mapview.max_lng = parseFloat(element.longitude);
+		album.photos.forEach(
+			/** @param {Photo} element */ function (element) {
+				// TODO: My IDE complains that `parseFloat` is unnecessary, because `element.latitude`/`element.longitude` is already a float
+				// TODO: My IDE complains that `element.size_variants.thumb !== null` is always false
```

Comment:
Probably correct about latitude/longitude (they can be `null` but that's already handled).

I've said about `thumb` earlier but since it keeps coming up, I just actually checked. It most definitely _is_ `null` when the thumb is missing for videos. Why does the IDE think that it can't be?

Reply:
Because the back-end code assumes that `thumb` is not null, too. As I said above you are completely right and the case of non-existing thumbs due to a missing FFmpeg never came to my mind. This might even lead to an overlooked bug in `SizeVariantFactory` of the back-end. I fear that I just assumed that FFmpeg is always there, oh, oh, ...

Reply:
We explicitly mark ffmpeg as optional on https://lycheeorg.github.io/docs/installation.html

Reply:
Now, after you have said that the scales fell from my eyes. But this doesn't mean I actively considered that when I wrote `SizeVariantFactory` for the back-end part. I am erring, too, from time to time :smile: